model/common/src/icon4py/model/common/orchestration/halo_exchange.py

```diff
@@ -0,0 +1,325 @@
+            fields_desc += f"{descr_unique_name} = std::make_unique<{descr_type_}>(*domain_descriptor, IN_field_{i}, levels, levels_first, outer_strides);\n"
+            fields_desc_glob_vars += f"std::unique_ptr<{descr_type_}> {descr_unique_name};\n"
+
+    code = ""
+    if counter == 0:
+        # The GHEX C++ ptrs are global variables
+        # Here, they are initialized at the first encounter of the halo exchange node
+        __pattern = ""
+        __domain_descriptor = ""
+        for dim_ in dims.global_dimensions.values():
+            __pattern += f"__pattern_{dim_.value}Dim_ptr_{unique_id} = IN___pattern_{dim_.value}Dim_ptr;\n"  # IN_XXX comes from the DaCe connector
+            __domain_descriptor += f"__domain_descriptor_{dim_.value}Dim_ptr_{unique_id} = IN___domain_descriptor_{dim_.value}Dim_ptr;\n"
+
+        code = f"""
+            __context_ptr_{unique_id} = IN___context_ptr;
+            __comm_ptr_{unique_id} = IN___comm_ptr;
+            {__pattern}
+            {__domain_descriptor}
+            """
+
+    code += f"""
+            ghex::context* m = reinterpret_cast<ghex::context*>(__context_ptr_{unique_id});
+            
+            {pattern_type}* pattern = reinterpret_cast<{pattern_type}*>(__pattern_{dim.value}Dim_ptr_{unique_id});
+            {domain_descriptor_type}* domain_descriptor = reinterpret_cast<{domain_descriptor_type}*>(__domain_descriptor_{dim.value}Dim_ptr_{unique_id});
+            {communication_object_type}* communication_object = reinterpret_cast<{communication_object_type}*>(__comm_ptr_{unique_id});
+
+            {fields_desc}
+
+            h_{unique_id} = communication_object->exchange({", ".join([f'(*pattern)({"" if wait else "*"}{descr_unique_names[i]})' for i in range(len(global_buffers))])});
+            { 'h_'+str(unique_id)+'.wait();' if wait else ''}
+
+            __out = 1234; // Dummy return;
+            """
+
+    tasklet.code = CodeBlock(code=code, language=dace.dtypes.Language.CPP)
+
+    if counter == 0:
+        # Set global variables
+        __pattern = ""
+        __domain_descriptor = ""
+        for dim_ in dims.global_dimensions.values():
+            __pattern += f"{dace.uintp.dtype} __pattern_{dim_.value}Dim_ptr_{unique_id};\n"
+            __domain_descriptor += (
+                f"{dace.uintp.dtype} __domain_descriptor_{dim_.value}Dim_ptr_{unique_id};\n"
+            )
+
+        code = f"""
+                {dace.uintp.dtype} __context_ptr_{unique_id};
+                {dace.uintp.dtype} __comm_ptr_{unique_id};
+                {__pattern}
+                {__domain_descriptor}
+                {fields_desc_glob_vars}
+                ghex::communication_handle<{communication_handle_type}> h_{unique_id};
+                """
+    else:
+        code = fields_desc_glob_vars
+
+    tasklet.code_global = CodeBlock(code=code, language=dace.dtypes.Language.CPP)
+
+
+@dace.library.environment
+class DaceGHEX:
```

Comment:
I not sure, but this whole class looks to me like it should be filled with information provided by the GHEX bindings instead of hardcoded here. Maybe a TODO here if it makes sense, to move this functionality to GHEX in the future